### PR TITLE
#13282: Fixes Android Studio Warnings under Kotlin section 

### PR DESCRIPTION
**Pull request by sonalyadav1**
**Fixes #13282**

Fixes the warnings in the Migration section under the Kotlin header.
(The list of warnings can be obtained via: Code -> Inspect Code.)

___________________________________________________________________________________________________________
**Tests performed** 
Tested ProdDebug on Pixel 4 Emulator with API level 30.
Verified that the app builds and runs successfully without introducing regressions.
___________________________________________________________________________________________________________
**BEFORE**
Screenshot 2024-12-19 at 12 00 27 PM
**AFTER**
Screenshot 2024-12-19 at 6 48 20 PM

### CODE REVIEW

**Comment by neeldoshii** [comment]
Strange, why didn't the workflow test didn't ran even though on force push it should have worked.

**Comment by nicolas-raoul** [comment]
@neeldoshii For newer contributors, a core developer has to "approve" the CI run. I guess it is GitHub's way of reducing costs.

**Comment by sonalyadav1** [comment]
@neeldoshii @nicolas-raoul Thanks for responding I will wait for your feedback.

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Would you mind fixing this test? Thanks!

NearbyParentFragmentPresenterTest > testSearchCloseToCurrentLocationWhenFar FAILED

**Comment by sonalyadav1** [comment]
will work on that and make the necessary changes as soon as possible.

**Review by sonalyadav1 — COMMENTED** [review]
@nicolas-raoul   FIXES 
**NearbyParentFragmentPresenterTest > testSearchCloseToCurrentLocationWhenFar FAILED**

**Comment by sonalyadav1** [comment]
Let me know if any update regarding changes .

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Unit test still failing:
```
NearbyParentFragmentPresenterTest > testSearchCloseToCurrentLocationWhenFar FAILED
    java.lang.AssertionError at NearbyParentFragmentPresenterTest.kt:309
```

**Comment by sonalyadav1** [comment]
Screenshot 2024-12-21 at 9 09 12 AM
I check this function individually before pushing

**Comment by sonalyadav1** [comment]
Screenshot 2024-12-21 at 9 13 23 AM
I run this test before pushing

**Comment by neeldoshii** [comment]
> I run this test before pushing


Check the commit test case review : https://github.com/commons-app/apps-android-commons/actions/runs/12417015780/job/34687407494

**Comment by sonalyadav1** [comment]
I think I messed up , I'll make new PR addressing the issue .

File: app/src/main/java/fr/free/nrw/commons/category/CategoryClient.kt
Comment (neeldoshii) [inline app/src/main/java/fr/free/nrw/commons/category/CategoryClient.kt:19]: try to avoid the changes which are not relevant to the PR(formating etc). This leads to increase in unrelated git diff which becomes harder for reviewers to review the changes.

File: app/build.gradle
Comment (neeldoshii) [inline app/build.gradle:89]: seems duplicate.

https://github.com/sonalyadav1/apps-android-commons/blob/6a98fa673020e690759d58354ddce35edca3262a/app/build.gradle#L51

File: app/src/main/java/fr/free/nrw/commons/category/CategoryDetailsActivity.kt
Comment (neeldoshii) [inline app/src/main/java/fr/free/nrw/commons/category/CategoryDetailsActivity.kt:220]: seems irrelevant can you revert this?

File: .idea/inspectionProfiles/Project_Default.xml
Comment (neeldoshii) [inline .idea/inspectionProfiles/Project_Default.xml:5]: revert this file.

**Comment by neeldoshii** [comment]
> I think I messed up , I'll make new PR addressing the issue .

its okay. you can even continue working on this PR. This will help in maintaining the logs of the PR.

**Comment by sonalyadav1** [comment]
@neeldoshii  Thanks for the suggestion! I see how keeping everything in one PR can help with maintaining logs, but I think it might be better to open a new PR for this. That way, it will be easier to keep things organized and ensure that each set of changes is focused on a specific issue. I’ll make a new PR and link it to this one for context. Let me know if you think that works.

**Pull request closed without merging**
